plan/logical_plan_builder.go

```diff
@@ -594,6 +594,11 @@ func (b *planBuilder) buildDistinct(child LogicalPlan, length int) *LogicalAggre
 	}
 	plan4Agg.SetChildren(child)
 	plan4Agg.SetSchema(child.Schema().Clone())
+	// Distinct will be rewritten as first_row, we reset the type here since the return type
+	// of first_row is not always the same as the column arg of first_row.
+	for i, col := range plan4Agg.schema.Columns {
```

Comment:
should we make the same change for all the places that create the schema of aggregate operator?

Reply:
We reset the type of aggregate here since this is not correct to set the schema of `plan4Agg` as the child's schema in line 596.

If the schema of aggregate is set correctly, we do not need to make the same change.